components/search/FacetDropdown.vue

```diff
@@ -0,0 +1,242 @@
+<template>
+  <b-dropdown
+    ref="dropdown"
+    :variant="dropdownVariant"
+    class="mr-2"
+    :data-type="facetType"
+    data-qa="search facet"
+  >
+    <template v-slot:button-content>
+      <span :data-qa="`${facetName} dropdown button`">
+        {{ facetName }}
+      </span>
+    </template>
+
+    <b-dropdown-form class="options-container">
+      <template v-if="isRadio">
+        <b-form-radio
+          v-for="(option, index) in sortOptions"
+          :key="index"
+          v-model="radioSelected"
+          :value="option"
+          :name="facet.name"
+          :data-qa="`${option} ${RADIO}`"
+          @change="applySelection"
+        >
+          {{ $t(`facets.${facet.name}.options.${option}`) }}
+        </b-form-radio>
+      </template>
+
+      <template v-else>
+        <b-form-checkbox
+          v-for="(option, index) in sortOptions"
+          :key="index"
+          v-model="preSelected"
+          :value="option.label"
+          :name="facet.name"
+          :data-qa="`${option.label} ${CHECKBOX}`"
+          :class="{ 'font-weight-bold' : selectedFacet.some(s => s === option.label) }"
+        >
+          {{ option.label }} ({{ option.count | localise }})
+        </b-form-checkbox>
+      </template>
+    </b-dropdown-form>
+
+    <li
+      v-if="facetType === 'checkbox'"
+      class="p-2 float-right"
+    >
+      <b-button
+        variant="link"
+        :disabled="!activateCheckboxResetButton"
+        @click="resetCheckboxSelection"
+      >
+        {{ $t('facets.button.reset') }}
+      </b-button>
+      <b-button
+        variant="primary"
+        :disabled="disableApplyButton"
+        :data-qa="`${facetName} apply button`"
+        @click.stop="applySelection"
+      >
+        {{ $t('facets.button.apply') }}
+      </b-button>
+    </li>
+
+    <li
+      v-else
+      class="p-2 float-right"
+    >
+      <b-button
+        variant="link"
+        :disabled="disableRadioResetButton"
+        @click.stop="resetRadioSelection"
+      >
+        {{ $t('facets.button.reset') }}
+      </b-button>
+    </li>
+  </b-dropdown>
+</template>
+
+<script>
+  import isEqual from 'lodash/isEqual';
+
+  export default {
+    props: {
+      facet: {
```

Comment:
Let's replace this prop with two, one for `name` (a string) and one for `fields` (an array).